### DISCUSSION TITLE
Apache2 base-path proxy

### DISCUSSION THREAD

**Discussion by vexcited** [coder/code-server#2224] [Q&A] [answered]
Hey,
it has been like 6hours i've been trying out to proxy code-server to my domain with base path /vscode/

here's my apache2 config

```
<VirtualHost *:80>
     ServerName sub.domain.ml
     DocumentRoot /var/www/domain
     ErrorLog ${APACHE_LOG_DIR}/error.log
     CustomLog ${APACHE_LOG_DIR}/access.log combined

     <Location /vscode/ >
          RewriteEngine On
          RewriteCond %{HTTP:Upgrade} =websocket
          RewriteRule /vscode/(.*)           ws://0.0.0.0:8888/$1 [P,L]
          RewriteCond %{HTTP:Upgrade} !=websocket
          RewriteRule /vscode/(.*)           http://0.0.0.0:8888/$1 [P,L]
          
          ProxyPreserveHost On
          ProxyPass http://0.0.0.0:8888/
          ProxyPassReverse http://0.0.0.0:8888/
     </Location>
</VirtualHost> 
```

i start code-server like this
`code-server --bind-addr 0.0.0.0:8888 --auth password --user-data-dir /var/lib/code-server`

i can access to my domain through domain.my/vscode/
i can login but after i got wss:// errors
`WebSocket connection to 'wss://domain.my/vscode/?type=Management&reconnectionToken=6b8f0816-7d5a-46b6-a889-bd4371981741&reconnection=false&skipWebSocketFrames=false' failed: Error during WebSocket handshake: Unexpected response code: 200`

i use cloudflare for my domain too if it help

**Comment by vexcited** [comment, marked as answer]
Oh ok, sorry i saw the problem

i removed those lines :
```
ProxyPreserveHost On
ProxyPass http://0.0.0.0:8888/
ProxyPassReverse http://0.0.0.0:8888/
```
and passed 0.0.0.0 to 127.0.0.1 if it helps..

now it works well ! hope it can help someone who having the same problem.

**Reply by bpmct** [reply]
thanks for marking the answer 😀
